rust/src/lib.rs

```diff
@@ -2142,24 +2142,25 @@ pub unsafe extern "C" fn zcashlc_propose_shielding(
                     })
             })?;
 
-        let from_addrs = match transparent_receiver.map_or_else(||
-            if account_receivers.len() > 1 {
-                Err(anyhow!(
-                    "Account has more than one transparent receiver with funds to shield; this is not yet supported by the SDK. Provide a specific transparent receiver to shield funds from."
-                ))
-            } else {
-                Ok(account_receivers.iter().next().map(|(a, v)| (*a, *v)))
-            },
-            |addr| Ok(account_receivers.get(&addr).map(|value| (addr, *value)))
-        )?.filter(|(_, value)| *value >= shielding_threshold) { Some((addr, _)) => {
-            [addr]
-        } _ => {
-            // There are no transparent funds to shield; don't create a proposal.
+        // If a specific receiver is specified, select only value for that receiver; otherwise,
+        // select value for all receivers.
```

Comment:
Document why we are selecting all receivers where previously we only selected one (because we're relying on the assumption that any transparent receivers with funds in them were either already linked together by transparent-only wallets, or are newly linked together by mobile SDK usage by choice for short-term usability, and that the transparent gap limits being used are low enough that we can separate these now-definitely-linked addresses from future unlinked addresses by raising the gap limit if/when we deploy a mechanism that enables unlinkable transparent detection).